nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -66,6 +68,30 @@ let
     #!${pkgs.runtimeShell}
     ${checkedSystemdBootBuilder} "$@"
     ${cfg.extraInstallCommands}
+    '';
+
+  loaderSettingsFormat = pkgs.formats.keyValue {
+    mkKeyValue = k: v: if v == null then "" else
+    lib.generators.mkKeyValueDefault { } " " k v;
+  };
+
+  loaderConfigFile = loaderSettingsFormat.generate "loader.conf" {
+    timeout = config.boot.loader.timeout;
+    console-mode = cfg.consoleMode;
+    editor = cfg.editor;
+    default = "nixos-*";
+  };
+
+  finalLanzabooteBuilder = pkgs.writeShellScript "install-lanzaboote.sh" ''
+    ${cfg.secureBoot.package}/bin/lzbt install \
+      --systemd ${config.systemd.package} \
+      --systemd-boot-loader-config ${loaderConfigFile} \
+      --public-key ${cfg.secureBoot.publicKeyFile} \
+      --private-key ${cfg.secureBoot.privateKeyFile} \
+      --configuration-limit ${toString (if cfg.configurationLimit == null then 0 else cfg.configurationLimit)} \
+      ${efi.efiSysMountPoint} \
+      /nix/var/nix/profiles/system-*-link
```

Comment:
I believe that the bootloader installer gets passed the toplevel of the current profile as an argument here, so this seems weird: https://github.com/NixOS/nixpkgs/blob/aa958deeb9542b516db9f71129fa166f208cb604/nixos/modules/system/activation/switch-to-configuration.pl#L101C12-L101C44

I know it's a super-edge-case for them to differ, but who's to say that system profiles really match the path-pattern that you gave?

It kinda seems like managing the limit on the number of profiles that `nixos-rebuild` manages somehow, as opposed to something that each and every bootloader installer does for itself. Perhaps there should be an extension/companion to bootspec where the various installers update a JSON manifest on the boot partition?

Reply:
Alternately, if it _is_ the right thing for the bootloader backends to manage this, it seems like each one should get passed a properly-lengthed list of profile toplevels (instead of just passing the latest one).

Reply:
cc @nikstur who wrote this code originally.

More specifically, we had harder problems like https://github.com/nix-community/lanzaboote/issues/135 too.

Anyway, 

> It kinda seems like managing the limit on the number of profiles should be something that nixos-rebuild does, instead of the bootloader installers. Perhaps there should be an extension/companion to bootspec where the various installers update a JSON manifest on the boot partition, to indicate what they've installed?

Potentially. This should be brought to the next Bootspec v2 discussion.

Reply:
I don't have a proper opinion on that yet and I feel like it's out of scope for this PR, would you open an issue to summarize this whole problem?

Reply:
Will do.